presto-main/src/test/java/com/facebook/presto/connector/system/TestSystemSession.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.system;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.common.transaction.TransactionId;
+import com.facebook.presto.common.type.VarcharType;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.security.ConnectorIdentity;
+import com.facebook.presto.spi.security.SelectedRole;
+import com.facebook.presto.spi.security.TokenAuthenticator;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slice;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+public class TestSystemSession
+{
+    @Test
+    public void testSystemSessionForExtraCredentials()
+    {
+        TransactionId transactionId = new TransactionId(UUID.randomUUID());
+
+        ConnectorTransactionHandle transactionHandle = new GlobalSystemTransactionHandle("connector-id", transactionId);
+
+        Optional<Principal> principal = Optional.of(() -> "principal");
+
+        Optional<SelectedRole> selectedRole = Optional.empty();
+
+        Map<String, TokenAuthenticator> extraAuthenticators = new HashMap<>();
+        extraAuthenticators.put("token1", new TokenAuthenticator() {
+            @Override
+            public Slice getAuthenticationToken(Slice authenticationTokenRequest)
+            {
+                return null;
+            }
+        });
```

Comment:
Why is this needed?

Reply:
Yeah, removed it since it’s not contributing to the behavior being tested.